plasmapy/particles/particle_class.py

```diff
@@ -421,9 +421,41 @@ class Particle(AbstractPhysicalParticle):
     ``'transition metal'``, ``'uncharged'``, and ``'unstable'``.
     """
 
-    @staticmethod
-    def _validate_arguments(argument, Z, mass_numb):
+    def __init__(
+        self,
+        argument: ParticleLike,
+        mass_numb: Integral = None,
+        Z: Integral = None,
+    ):
+        """Instantiate a |Particle| object and set private attributes."""
+
+        # If argument is a Particle instance, then construct a new
+        # Particle instance for the same particle.
+
+        if isinstance(argument, Particle):
+            argument = argument.symbol
+
+        self._inputs = argument, mass_numb, Z
+
+        self._initialize_attributes_and_categories()
+        self._store_particle_identity()
+        self._assign_particle_attributes()
+        self._add_charge_information()
+        self._add_half_life_information()
```

Comment:
Make double underscores for names

Reply:
@rocco8773 — It turns out that double leading underscores causes the interpreter to engage in [name mangling](https://docs.python.org/3/tutorial/classes.html#private-variables).  

> Any identifier of the form `__spam` (at least two leading underscores, at most one trailing underscore) is textually replaced with `_classname__spam`, where `classname` is the current class name with leading underscore(s) stripped.

I had no idea about this before, but that does explain a few things I've noticed in the past that seemed weird at the time.  Because of that, I'm going to leave these with single underscores, which follows the convention to mark these attributes as not being part of the "public" API.